### PR TITLE
feat: add op_geth

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -41,7 +41,8 @@ RUN foundryup
 
 RUN git clone https://github.com/ethereum-optimism/optimism.git && \
     cd optimism && \
-    git checkout tutorials/chain && \
+    git checkout develop && \
+    git pull origin develop && \
```
Comment (parithosh) [inline Dockerfile:45]: prolly worth using something like `git clone --depth 1 https://path/to/repo/foo.git -b bar`? so we do a shallow clone of just 1 branch?

Comment (barnabasbusa) [reply Dockerfile:45]: use the suggestions, so I can just merge with one click

File: src/contracts/contract_deployer.star [hunk 2]
```diff
@@ -63,13 +75,20 @@ def launch_contract_deployer(
...
                 "forge script scripts/Deploy.s.sol:Deploy --private-key $GS_ADMIN_PRIVATE_KEY --broadcast --rpc-url $L1_RPC_URL",
                 "sleep 3",
+                "CONTRACT_ADDRESSES_PATH=$DEPLOYMENT_OUTFILE forge script scripts/L2Genesis.s.sol:L2Genesis --sig 'runWithStateDump()' --chain-id $L2_CHAIN_ID",
                 "cd /workspace/optimism/op-node",
-                "go run cmd/main.go genesis l2 --deploy-config ../packages/contracts-bedrock/deploy-config/getting-started.json --l1-deployments ../packages/contracts-bedrock/deployments/getting-started/.deploy --outfile.l2 genesis.json --outfile.rollup rollup.json --l1-rpc $L1_RPC_URL",
-                "mkdir -p /network-configs",
-                "mv /workspace/optimism/op-node/genesis.json /network-configs/genesis.json",
-                "mv /workspace/optimism/op-node/rollup.json /network-configs/rollup.json",
-                "mv /workspace/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy /network-configs/.deploy",
+                "go run cmd/main.go genesis l2 \
```
Comment (parithosh) [inline src/contracts/contract_deployer.star:80]: rantings of a madlad, great job figuring this out!

File: src/el/op-geth/op_geth_launcher.star [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    image,
+    existing_el_clients,
+):
+    # log_level = input_parser.get_client_log_level_or_default(
+    #     participant_log_level, global_log_level, VERBOSITY_LEVELS
+    # )
+
+    network_name = shared_utils.get_network_name(launcher.network)
+
+    # el_min_cpu = int(el_min_cpu) if int(el_min_cpu) > 0 else EXECUTION_MIN_CPU
```
Comment (parithosh) [inline src/el/op-geth/op_geth_launcher.star:100]: don't wanna do memory limits?

Comment (barnabasbusa) [reply src/el/op-geth/op_geth_launcher.star:100]: not for v1

File: Dockerfile (hunk 1 above)
Comment (parithosh) [reply Dockerfile:45]: meh, v2 work

**Review by parithosh — APPROVED** [review]

**Pull request merged**